java/core/src/java/org/apache/orc/impl/WriterImpl.java

```diff
@@ -325,9 +327,9 @@ public boolean checkMemory(double newScale) throws IOException {
   }
 
   private boolean checkMemory() throws IOException {
-    if (rowsSinceCheck >= ROWS_PER_CHECK) {
+    long size = treeWriter.estimateMemory();
```

Comment:
If `estimateMemory` is called for each batch write, will the performance of the write be degraded?